### PR TITLE
Bump deps

### PR DESCRIPTION
**Pull request by wswoodruff**
- This will be a minor version bump since we're increasing support for React to be `16.8.x - v17.x.x`

- Sub-zero versioning means our next publish should be `0.4.0` => `0.4.1`

### CODE REVIEW

**Review by devinivy — APPROVED** [review]

**Comment by wswoodruff** [comment]
I'll squash these when the time comes to do any merging

File: package.json [hunk 1]
```diff
@@ -36,9 +36,9 @@
   "peerDependencies": {
     "redux": ">=3 <5",
     "redux-thunk": "2.x.x",
-    "immer": ">=1 <7",
+    "immer": ">=1 <=8.x.x",
```
Comment (devinivy) [inline package.json]: Can we stick with the style used for the other deps?
```suggestion
    "immer": ">=1 <9",
```

File: package.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
     "@rollup/plugin-node-resolve": "7.x.x",
     "@testing-library/react-hooks": "3.x.x",
     "coveralls": "3.x.x",
-    "immer": "6.x.x",
+    "immer": "8.x.x",
     "normalizr": "3.x.x",
     "npm-run-all": "4.x.x",
     "react": "^16.8.x",
```
Comment (devinivy) [inline package.json]: Hey @wswoodruff it looks like react isn't being tested on v17: could we ensure we get those tests to run?

File: package.json [hunk 3]
```diff
@@ -48,11 +48,11 @@
...
     "@testing-library/react-hooks": "3.x.x",
     "coveralls": "3.x.x",
-    "immer": "6.x.x",
+    "immer": "8.x.x",
     "normalizr": "3.x.x",
     "npm-run-all": "4.x.x",
-    "react": "^16.8.x",
-    "react-test-renderer": "^16.8.x",
+    "react": "17.x.x",
+    "react-test-renderer": "16.8.x",
```
Comment (devinivy) [inline package.json]: Hey @wswoodruff react-test-renderer v16 is meant to work with react v16.  I believe this should be on v17.

**Review by devinivy — APPROVED** [review]

**Review by devinivy — APPROVED** [review]

**Comment by zemccartney** [comment]
Closing because extended by #16 🙏

**Pull request closed without merging**